packages/x-date-pickers/src/TimeClock/Clock.tsx

```diff
@@ -325,27 +334,27 @@ export function Clock(inProps: ClockProps) {
     switch (event.key) {
       case 'Home':
         // reset both hours and minutes
-        handleValueChange(0, 'partial');
+        handleValueChange(minViewValue, 'partial');
         event.preventDefault();
         break;
       case 'End':
-        handleValueChange(type === 'minutes' ? 59 : 23, 'partial');
+        handleValueChange(maxViewValue, 'partial');
         event.preventDefault();
         break;
       case 'ArrowUp':
-        handleValueChange(viewValue + keyboardControlStep, 'partial');
+        handleValueChange(circleValue(viewValue + keyboardControlStep), 'partial');
```

Comment:
Nitpick: It's slightly annoying that this component does not jump over disabled values with arrow navigation. 🙈 
However, we recommend it for mobile use, so, I'm not sure if it's even worth treating as a problem. 🤷